src/profilePage/ProfilePage.tsx

```diff
@@ -29,16 +36,31 @@ interface RouteParams {
 }
 
 function ProfilePage() {
+  const navigate = useNavigate({ from: "/" });
+
   const { profileId } = Route.useParams<RouteParams>();
   const profileName = profileId;
 
   //tanstack/react-query hook to fetch the users
-  const { data: profileData } = useQuery({
+  const {
+    data: profileData,
+    isError: isProfileError,
+    isPending,
+    isLoading,
+  } = useQuery({
     queryKey: [profileId],
     queryFn: () => getProfile(profileName),
   });
 
-  let { data: repoData } = useQuery({
+  //This does not work I will have to ask for advice on how to properly implement this
+  //problem: still routes when loader is fetched (I know why, I don't know how to fix it)
+  // if (!profileData && (isPending || isLoading)) {
+  //   setTimeout(function () {
+  //     navigate({ to: "/noData" });
```

Comment:
Have you tried just passing the reference of `navigate` with an arrow function instead of calling it. 
If I understand correctly, I think this might be because if you have `fun(fun2())`, `fun2` is executed first and its return value is passed to `fun`. So in your case fun2 = navigate and function is fun...